hazelcast/src/main/java/com/hazelcast/map/impl/proxy/NearCachedMapProxyImpl.java

```diff
@@ -295,7 +318,8 @@ protected void removeAllInternal(Predicate predicate) {
     }
 
     @Override
-    protected void deleteInternal(Data key) {
+    protected void deleteInternal(Object key) {
+        key = serializeKeys ? toDataWithStrategy(key) : key;
```

Comment:
can we do this a single method: `serializeKeys ? toDataWithStrategy(key) : key`?